src/user/dto/update-user-request.ts

```diff
@@ -0,0 +1,12 @@
+import { Role } from '@prisma/client';
+
+export interface UpdateUserRequest {
+  where: {
+    id: number;
+  };
+  data: {
+    email?: string;
+    password?: string;
+    role?: Role;
+  };
+}
```

Comment:
shouldn't this be a class with `class-validator` validation applied?

Reply:
Right, to be consistent it should not have such name, but `UpdateUserRequestDto` is kind of counter intuitive for me because I update nothing - I just wanna map fields and return something else. What about `MappedUserRequest` or `UserPayloadRequest`